src/controllers/match.ts

```diff
@@ -71,3 +72,69 @@ export const match = async (
     next(new InternalServerError())
   }
 }
+
+export const employerMatch = async (
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => {
+  try {
+    const employer = req.user as Employer
+    const postId = (req as any).postId
+    if (!employer) return next(new NotFoundError('User not found'))
+
+    const post = await JobPost.findOne(postId)
+    if (!post) return next(new NotFoundError('Job Post not found'))
+    const skillsInPost = post.skills.map((skill) => skill.id)
+
+    const jobseeker = (await Promise.all(
+      skillsInPost.map(async (id) => {
+        try {
+          const skill = (await Skill.findOne(id, {
+            relations: ['jobSeekers'],
+          })) as Skill
+
+          if (skill.jobSeekers.length === 0) return
+          return skill.jobSeekers
+        } catch (error) {
+          next(new InternalServerError())
+        }
+      })
+    )) as JobSeeker[][]
+
+    //**Flaten the array of jobseekers
+    const matchedJobseekers = _.flatten(jobseeker)
+    if (matchedJobseekers.length === 0)
+      return next(new NotFoundError('No match found'))
+
+    //**Count the times that a jobseeker come up and store it in a object*/
+    //**The count is equivalent to how many skills is matched */
+    const matchCount = matchedJobseekers.reduce(
```

Comment:
now understood, why employer always expect good data structure in programming language :))